src/app/calculation/scores.py

```diff
@@ -44,55 +50,52 @@ def calculate_scores_infra_types(infra_type, cur, conn):
+            f'avg_c_score, ' \
             f'avg_incident_count, ' \
-            f'scary_incident_count, ' \
             f'avg_scary_incident_count, ' \
-            f'c_count, ' \
             f'avg_c_count, ' \
-            f'a_count, ' \
             f'avg_a_count ' \
             f'from (' \
             f'select ' \
             f'sum(count) as count, ' \
-            f'round(avg(a_score), 4) as avg_a_score, ' \
-            f'round(avg(c_score), 4) as avg_c_score, ' \
-            f'sum("chosenCount") as c_count, ' \
-            f'sum("avoidedCount") as a_count, ' \
-            f'sum("normalIncidentCount") as incident_count, ' \
+            f'case when (sum(count)::numeric != 0) then round(sum("avoidedCount")::numeric / sum(count)::numeric, 4) else 1 end as avg_a_score, ' \
+            f'case when (sum(count)::numeric != 0) then round(sum("chosenCount")::numeric / sum(count)::numeric, 4) else 0 end as avg_c_score, ' \
             f'round(avg("normalIncidentCount"), 4) as avg_incident_count, ' \
-            f'sum("scaryIncidentCount") as scary_incident_count, ' \
             f'round(avg("scaryIncidentCount"), 4) as avg_scary_incident_count, ' \
             f'round(avg("chosenCount"), 4) as avg_c_count, ' \
             f'round(avg("avoidedCount"), 4) as avg_a_count, ' \
-            f'round(avg(danger_score)) as avg_danger_score, ' \
-            f'round(sum("chosenCount")::numeric / (sum("avoidedCount")::numeric + sum("chosenCount")::numeric), 4)::numeric as p_score, ' \
-            f'round(greatest(least(1 - (1 / sum(count * (ST_Length(geom::geography)::numeric / 1000))::numeric * (4.4 * sum("scaryIncidentCount")::numeric + sum("normalIncidentCount")::numeric)), 1)::numeric, 0)::numeric, 4)::numeric as s_score ' \
-            f'from "SimRaAPI_osmwayslegs" ' \
-            f"where '{infra_type}' = any(infra_type) " \
-            f'and (count > 0 or "avoidedCount" > 0)' \
+            f'round((1 / sum(count * (ST_Length(geom::geography)::numeric / 1000))::numeric * (4.4 * sum("scaryIncidentCount")::numeric + sum("normalIncidentCount")::numeric)), 4) as avg_danger_score, ' \
+            f'case when((sum("avoidedCount")::numeric + sum("chosenCount")::numeric) != 0) ' \
+            f'then round(sum("chosenCount")::numeric / (sum("avoidedCount")::numeric + sum("chosenCount")::numeric), 4)::numeric ' \
+            f'else -1 ' \
+            f'end as p_score, ' \
+            f'case when (sum(count * (ST_Length(geom::geography)::numeric / 1000))::numeric != 0) ' \
+            f'then round(greatest(least(1 - (1 / sum(count * (ST_Length(geom::geography)::numeric / 1000))::numeric * (4.4 * sum("scaryIncidentCount")::numeric + sum("normalIncidentCount")::numeric)), 1)::numeric, 0)::numeric, 4)::numeric ' \
+            f'else -1 ' \
+            f'end as s_score ' \
+            f'from "SimRaAPI_osmwayslegsused" ' \
+            f"where '{infra_type}' = any(infra_type)" \
+            f'and (count > 0 or "avoidedCount" > 0) '\

```

Comment:
Can be omitted as init_smaller_table is used

Reply:
yes line 90 could be omitted. (we still need the rest)